kubernetes-itests/src/test/java/io/fabric8/kubernetes/DeploymentIT.java

```diff
@@ -128,6 +128,6 @@ public void cleanup() throws InterruptedException {
     Thread.sleep(6000);
     client.extensions().deployments().inNamespace(currentNamespace).delete();
     // Wait for resources to get destroyed
-    Thread.sleep(30000);
+    Thread.sleep(60000);
```

Comment:
How come 60000?

Reply:
Just a guess that 1 minute will be good, and also CI went green. So

Reply:
@piyush1594 Do not merge yet. Is there a way to query if namespace is deleted? Something like `client.extensions.inNamespace.status`

Reply:
Hey @lordofthejars 

You can try

```
client.namespaces().list();
```
and
```
client.namespaces().withName("").get();
```

Reply:
Ok, then look we can use awaitility instead of a thread.sleep so you can do:
https://github.com/awaitility/awaitility

```java
await().atMost(2, MINUTE).until(namespaceDeleted());
```

So this internally implements a poling so it will not take all the time one minute but just the amount of time we need. So we can put 2 minutes for just in case there are some slow machines.

Reply:
so here we are deleting the deployment and pod, so we need to put that condition @lordofthejars

Reply:
You need to replace the `Thread.sleep` for the `await()` call

Reply:
agreed with @lordofthejars.